tests/test_web_poet_rules.py

```diff
@@ -0,0 +1,1173 @@
+
+    item, deps = yield crawl_item_and_deps(ParentReplacedProductPage)
+    assert item == ParentReplacedProduct(name="parent replaced product name")
+    assert_deps(deps, {"page": ParentReplacedProductPage})
+
+    item, deps = yield crawl_item_and_deps(SubclassReplacedProductPage)
+    assert item == ParentReplacedProduct(name="subclass replaced product name")
+    assert_deps(deps, {"page": SubclassReplacedProductPage})
+
+
+@attrs.define
+class StandaloneProduct:
+    name: str
+
+
+@handle_urls(URL, to_return=StandaloneProduct)
+class StandaloneProductPage(ItemPage):
+    @field
+    def name(self) -> str:
+        return "standalone product name"
+
+
+@inlineCallbacks
+def test_item_to_return_standalone(caplog) -> None:
+    """Same case as with ``test_item_to_return_in_handle_urls()`` above but the
+    page object doesn't inherit from something like ``ItemPage[ItemClass]``
+    """
+    item, deps = yield crawl_item_and_deps(StandaloneProduct)
+    assert "UndeclaredProvidedTypeError:" in caplog.text
+    assert item is None
+    assert_deps(deps, {}, size=0)
+
+    # calling the actual page object should still work
+    item, deps = yield crawl_item_and_deps(StandaloneProductPage)
+    assert item == {"name": "standalone product name"}
+    assert_deps(deps, {"page": StandaloneProductPage})
+
+
+@attrs.define
+class Morty:
+    name: str
+
+
+@handle_urls(URL)
+class RickSanchezPage(ItemPage[Morty]):
+    @field
+    def name(self) -> str:
+        return "from basic rick"
+
+
+@handle_urls(URL, instead_of=RickSanchezPage)
+class RickSanchezC137Page(ItemPage[Morty]):
+    @field
+    def name(self) -> str:
+        return "wubba lubba dub dub"
+
+
+@inlineCallbacks
+def test_item_return_with_overrides() -> None:
+    item, deps = yield crawl_item_and_deps(Morty)
+    assert item == Morty(name="wubba lubba dub dub")
+    assert_deps(deps, {"item": RickSanchezC137Page})
+
+    # page from 'instead_of'
+    item, deps = yield crawl_item_and_deps(RickSanchezPage)
+    assert item == Morty(name="wubba lubba dub dub")
+    assert_deps(deps, {"page": RickSanchezC137Page})
+
+    # calling the actual page object should still work
+    item, deps = yield crawl_item_and_deps(RickSanchezC137Page)
+    assert item == Morty(name="wubba lubba dub dub")
+    assert_deps(deps, {"page": RickSanchezC137Page})
+
+
+@attrs.define
+class ProductFromInjectable:
+    name: str
+
+
+@handle_urls(URL, to_return=ProductFromInjectable)
+class ProductFromInjectablePage(Injectable):
+    @field
+    def name(self) -> str:
+        return "product from injectable"
+
+    async def to_item(self) -> ProductFromInjectable:
+        return await item_from_fields(self, item_cls=ProductFromInjectable)
+
+
+@inlineCallbacks
+def test_item_return_from_injectable() -> None:
+    """The case wherein a PageObject inherits directly from ``web_poet.Injectable``
+    should also work, provided that the ``to_item`` method is similar with
+    ``web_poet.ItemPage``:
```

Comment:
I added them here to function as regression checks to err out if something changes in the future. Leaving it as-is here since we get support on it for free with the current implementation. Although I doubt users are going to have this use case.